src/helpers.tsx

```diff
@@ -0,0 +1,21 @@
+import { Todo } from './types/Todo';
+
+export const filterTodos = (
+  todos: Todo[],
+  filter: string,
+  query: string,
+) => {
+  return todos.filter(todo => {
+    switch (filter) {
+      case 'completed':
+        return todo.completed;
+      case 'active':
+        return !todo.completed;
+      default:
+        return todo;
+    }
+  })
+    .filter(todo => (
+      todo.title.toLowerCase().includes(query.toLowerCase().trim())
+    ));
```

Comment:
1. You create `query.toLowerCase().trim()` every iteration 
2. You can move this logic to the first filter instead of creating a new one